atlasdb-cassandra-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceTest.java

```diff
@@ -75,4 +83,30 @@ public void testCreateTableCaseInsensitive() {
         Preconditions.checkArgument(!allTables.contains(table3));
     }
 
+    @Test
+    public void testLockLeaderCreatesLockTable() {
+        CassandraKeyValueService kvs = CassandraKeyValueService.create(
+                CassandraKeyValueServiceConfigManager.createSimpleManager(
+                        CassandraTestSuite.CASSANDRA_KVS_CONFIG.withKeyspace("lockLeader")),
+                        CassandraTestSuite.LEADER_CONFIG);
+
+        assertThat(kvs.getLockTable(), is(true));
+    }
+
+    @Test
+    public void testNonLockLeaderDoesNotCreateLockTable() throws InterruptedException, ExecutionException, TimeoutException {
+        Future async = CassandraTestTools.async(executorService, this::createKvsAsNonLockLeader);
+
+        Thread.sleep(5*1000);
+
+        CassandraTestTools.assertThatFutureDidNotSucceedYet(async);
```

Comment:
I'm pretty sure it won't flake. We set up a new keyspace and have one CKVS object (representing one Atlas server) that is not the lock leader. It will wait forever for the lock leader to appear.